tests/Unit/Core/Form/IdentifiableObject/CommandBuilder/Product/ProductAttachmentsCommandsBuilderTest.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace Tests\Unit\Core\Form\IdentifiableObject\CommandBuilder\Product;
+
+use Generator;
+use PrestaShop\PrestaShop\Core\Domain\Product\Attachment\Command\RemoveAllAssociatedProductAttachmentsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Attachment\Command\SetAssociatedProductAttachmentsCommand;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Product\ProductAttachmentsCommandsBuilder;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Product\ProductCommandsBuilderInterface;
+
+class ProductAttachmentsCommandsBuilderTest extends AbstractProductCommandBuilderTest
+{
+    /**
+     * @var ProductCommandsBuilderInterface
+     */
+    private $commandsBuilder;
+
+    protected function setUp(): void
+    {
+        $this->commandsBuilder = new ProductAttachmentsCommandsBuilder();
+    }
+
+    /**
+     * @dataProvider getExpectedCommands
+     *
+     * @param array $formData
+     * @param array $expectedCommands
+     */
+    public function testBuildCommand(array $formData, array $expectedCommands): void
+    {
+        $builtCommands = $this->commandsBuilder->buildCommands($this->getProductId(), $formData);
+        $this->assertEquals($expectedCommands, $builtCommands);
+    }
+
+    public function getExpectedCommands(): Generator
+    {
+        yield [
```

Comment:
Nice-to-have: I believe there must be a way to have a single `yield` statement

Reply:
Then there would be no interest in using `yield` The principle of `yield` is to have an iterable result, each `yield` symbolizing a step of the iterator If there's only one `yield` it means there's only one step so no need for a dataProvider

Usually, the data providers return an array, each first level being one set of data for one execution of the test
This results in some hard to read dataProvider methods, that's why we started using `yield` (at least that's my reason) because this way each tests is clearly identified